pkg/platform/coverage/javascript/javascript.go

```diff
@@ -9,31 +9,51 @@ import (
 	"os/exec"
 	"path/filepath"
 	"strconv"
+	"strings"
 
 	"go.keploy.io/server/v2/pkg/models"
 	"go.keploy.io/server/v2/pkg/platform/coverage"
+	"go.keploy.io/server/v2/utils"
 	"go.uber.org/zap"
 )
 
 type Javascript struct {
-	ctx      context.Context
-	logger   *zap.Logger
-	reportDB coverage.ReportDB
-	cmd      string
+	ctx         context.Context
+	logger      *zap.Logger
+	reportDB    coverage.ReportDB
+	cmd         string
+	commandType string
 }
 
-func New(ctx context.Context, logger *zap.Logger, reportDB coverage.ReportDB, cmd string) *Javascript {
+func New(ctx context.Context, logger *zap.Logger, reportDB coverage.ReportDB, cmd, commandType string) *Javascript {
 	return &Javascript{
-		ctx:      ctx,
-		logger:   logger,
-		reportDB: reportDB,
-		cmd:      cmd,
+		ctx:         ctx,
+		logger:      logger,
+		reportDB:    reportDB,
+		cmd:         cmd,
+		commandType: commandType,
 	}
 }
 
 func (j *Javascript) PreProcess() (string, error) {
+	err := os.Setenv("CLEAN", "true")
+	if err != nil {
```

Comment:
Will these changes work for docker-compose as well ?

Reply:
I will test with docker compose once

Reply:
no extra changes required to support docker compose workflow for getting coverage

the docker compose workflow does not depend on preprocess function call. we need to edit docker-compose file and add the volumes and env variables that we are adding in case of  docker run command in preprocess function.

Reply:
As discussed, we would be writing volume source:destination pair and env variables in docker-compose-tmp.yaml file and running `docker compose up` with it.

we are doing it to remove the manual step of user editing their docker compose yaml file and simplying user experience.

https://github.com/keploy/keploy/pull/2064/commits/97eb9371aa12cb4eb2e8630b7b7a7592d9a2c5e9